drivers/net/wireless/marvell/mwifiex/sta_cmdresp.c

```diff
@@ -708,6 +708,23 @@ static int mwifiex_ret_ver_ext(struct mwifiex_private *priv,
 {
 	struct host_cmd_ds_version_ext *ver_ext = &resp->params.verext;
 
+	if (priv->ds_quirk_version_string_requested) {
+		if (strncmp(ver_ext->version_str, "ChipRev:20, BB:9b(10.00), RF:40(21)", 128) == 0) {
+			struct mwifiex_ds_auto_ds auto_ds = {
+				.auto_ds = DEEP_SLEEP_OFF,
+			};
+
+			mwifiex_dbg(priv->adapter, MSG,
+				    "Bad HW revision detected, disabling deep sleep\n");
+
+			mwifiex_send_cmd(priv, HostCmd_CMD_802_11_PS_MODE_ENH,
+					 DIS_AUTO_PS, BITMAP_AUTO_DS, &auto_ds, false);
+		}
+
+		priv->ds_quirk_version_string_requested = false;
+		return 0;
+	}
```

Comment:
Does this need some sort of synchronization? If so, maybe replace that with atomic bitops, e.g. `set_bit()` in `maybe_quirk_fw_disable_ds()` and `test_and_clear_bit()` here.

I guess this should work as is though as `mwifiex_ret_ver_ext()` and `maybe_quirk_fw_disable_ds()` both shouldn't be entered multiple times in parallel?

Reply:
I think the only risk here is that a user-invoked (from sysfs) verext command gets queued right before we queue ours. Then the response handler for the user-invoked command will get called first and `ds_quirk_version_string_requested` will be set to `false`. Not sure if we want to care about that..

But yeah, using `set_bit()` and `test_and_clear_bit()` sounds good indeed.

Reply:
Right, I think with `set_bit()` and `test_and_clear_bit()` you don't have to think about any such stuff and it should "just work".

Reply:
Alright, switched to using a bit now